frontend/packages/dev-console/src/components/import/deploy-image/SearchField.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+
+import { k8sCreate } from '@console/internal/module/k8s';
+import { ImageStreamImportsModel } from '@console/internal/models';
+import { getPorts } from '@console/internal/components/source-to-image';
+import { useFormikContext, FormikValues } from 'formik';
+import { InputSearchField } from '../../formik-fields';
+
+const getSuggestedName = (name: string): string => {
+  if (!name) {
+    return;
+  }
+
+  const imageName: string = _.last(name.split('/'));
+
+  return _.first(imageName.split(/[^a-z0-9-]/)); /* eslint-disable-line consistent-return */
```

Comment:
We shouldn't be disabling rules.

Reply:
do you know how to get past this one? it is for JS not TS.

Reply:
line 13 has no return value. If you want to support `undefined`, explicitly return `undefined`

Reply:
I want to either return the name or stop trying if there is no name to work with.